core/src/main/java/com/google/cloud/sql/CredentialFactory.java

```diff
@@ -17,11 +17,12 @@
 package com.google.cloud.sql;
 
 import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.http.HttpRequestInitializer;
 
 /** Factory for creating {@link Credential}s for interaction with Cloud SQL Admin API. */
 public interface CredentialFactory {
   /** Name of system property that can specify an alternative credential factory. */
   String CREDENTIAL_FACTORY_PROPERTY = "cloudSql.socketFactory.credentialFactory";
 
-  Credential create();
+  HttpRequestInitializer create();
```

Comment:
This is a breaking changing to a public interface.
I would expect a major version bump for this to permissible.

Reply:
It might break binary compatibility, but it's source compatible

Reply:
I didn't realize `Credential` implements `HttpRequestInitializer`.  This is actually OK then.